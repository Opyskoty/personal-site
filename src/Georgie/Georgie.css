.Georgie {
  margin-top: 70px !important;
  text-align: center;
  align-items: center;
  padding-bottom: 10px;
  padding-top: 10px;
  font-family: "Inconsolata", monospace;
  font-size: 1.2em;
  border: 1.5px solid rgb(163, 245, 243) !important;
}

.carousel {
  width: 50%;
}

.carousel-item > img {
  width: 90%;
  height: 500px;
  border-radius: 30px!important;
}
