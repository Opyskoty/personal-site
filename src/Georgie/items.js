import georgie1 from "./georgie1.jpg";
import georgie2 from "./georgie2.png";
import georgie3 from "./georgie3.JPG";
import georgie4 from "./georgie4.JPG";
import georgie5 from "./georgie5.jpg";
import georgie6 from "./georgie6.jpg";
import georgie7 from "./georgie7.JPG";

const items = [
  {
    src: georgie1,
    altText: "Georgie 1",
  },
  {
    src: georgie2,
    altText: "Georgie 2",
  },
  {
    src: georgie3,
    altText: "Georgie 3",
  },
  {
    src: georgie4,
    altText: "Georgie 4",
  },
  {
    src: georgie5,
    altText: "Georgie 5",
  },
  {
    src: georgie6,
    altText: "Georgie 6",
  },
  {
    src: georgie7,
    altText: "Georgie 7",
  },
];


export default items;
