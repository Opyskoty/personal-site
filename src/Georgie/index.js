import React, { useState } from "react";
import "./Georgie.css";
import {
  Container,
  Row,
  Col,
  Carousel,
  CarouselItem,
  CarouselControl,
  CarouselIndicators,
  CarouselCaption,
} from "reactstrap";
import georgie1 from "./georgie1.jpg";
import georgie2 from "./georgie2.png";
import georgie3 from "./georgie3.JPG";
import georgie4 from "./georgie4.JPG";
import georgie5 from "./georgie5.jpg";
import georgie6 from "./georgie6.jpg";
import georgie7 from "./georgie7.JPG";

const items = [
  {
    src: georgie1,
    altText: "Georgie 1",
  },
  {
    src: georgie2,
    altText: "Georgie 2",
  },
  {
    src: georgie3,
    altText: "Georgie 3",
  },
  {
    src: georgie4,
    altText: "Georgie 4",
  },
  {
    src: georgie5,
    altText: "Georgie 5",
  },
  {
    src: georgie6,
    altText: "Georgie 6",
  },
  {
    src: georgie7,
    altText: "Georgie 7",
  },
];

function Georgie() {
  const [activeIndex, setActiveIndex] = useState(0);
  const [animating, setAnimating] = useState(false);

  const next = () => {
    if (animating) return;
    const nextIndex = activeIndex === items.length - 1 ? 0 : activeIndex + 1;
    setActiveIndex(nextIndex);
  };

  const previous = () => {
    if (animating) return;
    const nextIndex = activeIndex === 0 ? items.length - 1 : activeIndex - 1;
    setActiveIndex(nextIndex);
  };

  const goToIndex = (newIndex) => {
    if (animating) return;
    setActiveIndex(newIndex);
  };

  const slides = items.map((item) => {
    return (
      <CarouselItem
        className="carouselImg"
        onExiting={() => setAnimating(true)}
        onExited={() => setAnimating(false)}
        key={item.src}
      >
        <img src={item.src} alt={item.altText} />
      </CarouselItem>
    );
  });

  return (
    <Container className="Georgie">
      <Row>
        <Col>
          <h3>Bonus Dog Content:</h3>
          <p>
            Although not a software developer, my dog Georgie is incredibly
            adorable and I thought that warranted having an entire page of my
            website dedicated to her. Enjoy some of her cutest moments:
          </p>
        </Col>
      </Row>
      <Row>
        <Col
          className="flexBox"
          style={{
            display: "flex",
            flexFlow: "row wrap",
            justifyContent: "center",
          }}
        >
          <Carousel
            className="carousel"
            activeIndex={activeIndex}
            next={next}
            previous={previous}
          >
            <CarouselIndicators
              items={items}
              activeIndex={activeIndex}
              onClickHandler={goToIndex}
            />
            {slides}
            <CarouselControl
              direction="prev"
              directionText="Previous"
              onClickHandler={previous}
            />
            <CarouselControl
              direction="next"
              directionText="Next"
              onClickHandler={next}
            />
          </Carousel>
        </Col>
      </Row>
    </Container>
  );
}

export default Georgie;
