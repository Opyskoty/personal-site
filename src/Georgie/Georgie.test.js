import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Georgie from "./index";

it("renders without crashing", function () {
  render(<Georgie />);
});

it("displays the next picture when clicking on arrow", function () {
  const { getByText, getByAltText } = render(<Georgie />);

  const prevButton = getByText("Previous");
  const nextButton = getByText("Next");

  fireEvent.click(nextButton);

  expect(getByAltText("Georgie 2")).toBeInTheDocument();

  fireEvent.click(prevButton);

  expect(getByAltText("Georgie 7")).toBeInTheDocument();
});

//snapshot
it("matches snapshot", function () {
  const { asFragment } = render(<Georgie />);
  expect(asFragment()).toMatchSnapshot();
});
