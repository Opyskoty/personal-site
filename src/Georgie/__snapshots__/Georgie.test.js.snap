// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Georgie container"
  >
    <div
      class="row"
    >
      <div
        class="col"
      >
        <h3>
          Bonus Dog Content:
        </h3>
        <p>
          Although not a software developer, my dog Georgie is incredibly adorable and I thought that warranted having an entire page of my website dedicated to her. Enjoy some of her cutest moments:
        </p>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="flexBox col"
        style="display: flex; flex-flow: row wrap; justify-content: center;"
      >
        <div
          class="carousel carousel slide"
        >
          <ol
            class="carousel-indicators"
          >
            <li
              class="active"
            />
            <li
              class=""
            />
            <li
              class=""
            />
            <li
              class=""
            />
            <li
              class=""
            />
            <li
              class=""
            />
            <li
              class=""
            />
          </ol>
          <div
            class="carousel-inner"
          >
            <div
              class="carouselImg carousel-item active"
            >
              <img
                alt="Georgie 1"
                src="georgie1.jpg"
              />
            </div>
            <div
              class="carouselImg carousel-item"
            >
              <img
                alt="Georgie 2"
                src="georgie2.png"
              />
            </div>
            <div
              class="carouselImg carousel-item"
            >
              <img
                alt="Georgie 3"
                src="georgie3.JPG"
              />
            </div>
            <div
              class="carouselImg carousel-item"
            >
              <img
                alt="Georgie 4"
                src="georgie4.JPG"
              />
            </div>
            <div
              class="carouselImg carousel-item"
            >
              <img
                alt="Georgie 5"
                src="georgie5.jpg"
              />
            </div>
            <div
              class="carouselImg carousel-item"
            >
              <img
                alt="Georgie 6"
                src="georgie6.jpg"
              />
            </div>
            <div
              class="carouselImg carousel-item"
            >
              <img
                alt="Georgie 7"
                src="georgie7.JPG"
              />
            </div>
          </div>
          <a
            class="carousel-control-prev"
            role="button"
            style="cursor: pointer;"
            tabindex="0"
          >
            <span
              aria-hidden="true"
              class="carousel-control-prev-icon"
            />
            <span
              class="sr-only"
            >
              Previous
            </span>
          </a>
          <a
            class="carousel-control-next"
            role="button"
            style="cursor: pointer;"
            tabindex="0"
          >
            <span
              aria-hidden="true"
              class="carousel-control-next-icon"
            />
            <span
              class="sr-only"
            >
              Next
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
